From 6bc05b0a27f034221c96216b6ca78f6854f308df Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Sun, 28 Mar 2021 23:13:01 -0400
Subject: [PATCH] arm32: kaslr: Fix the bug of module install failure
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6bc05b0a27f034221c96216b6ca78f6854f308df


hulk inclusion
category: bugfix
bugzilla: 47952
CVE: NA

------------------------------------------------------------------------

Linux can't enable fpic to compile modules, because the modules
have their own relocation table, and they can't use the got table
for symbolic addressing.

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 4f550fc0e811..cb7c6b02fede 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -50,6 +50,7 @@ endif
 
 ifeq ($(CONFIG_RELOCATABLE),y)
 KBUILD_CFLAGS		+= -fpic -include $(srctree)/include/linux/hidden.h
+CFLAGS_MODULE		+= -fno-pic
 LDFLAGS_vmlinux		+= -pie -shared -Bsymbolic
 endif
 
-- 
2.26.2

